pkg/apis/logging/v1/elasticsearch_types.go

```diff
@@ -59,11 +60,32 @@ type ElasticsearchSpec struct {
 	// Important: Run "operator-sdk generate k8s" to regenerate code after modifying this file
 	// Add custom validation using kubebuilder tags: https://book.kubebuilder.io/beyond_basics/generating_crd.html
 
-	ManagementState  ManagementState       `json:"managementState"`
-	RedundancyPolicy RedundancyPolicyType  `json:"redundancyPolicy"`
-	Nodes            []ElasticsearchNode   `json:"nodes"`
-	Spec             ElasticsearchNodeSpec `json:"nodeSpec"`
-	IndexManagement  *IndexManagementSpec  `json:"indexManagement"`
+	// Indicator if the resource is 'Managed' or 'Unmanaged' by the operator
+	//
+	// +kubebuilder:validation:Enum:=Managed;Unmanaged
+	// +optional
```

Comment:
technically not optional because we have the enum

Reply:
Technically correct, but we didn't enforce this on API side validation. In turn what if we have already stored objects w/o ManagementState field set, which the API allows but our operator just don't process anymore. Is this a breaking change?

Reply:
Actually testing this with a persistent object w/o ManagementState and the new CRD generation has the same effect as previously. It produces error logs on the operator side.